src/Controller/QuizTemplateController.php

```diff
@@ -39,14 +42,25 @@ public function createQuiz(Request $request)
 
     public function showQuizzes(Request $request, array $requestAttributes)
     {
-        $arguments['currentPage'] = (int)$request->getParameter('page');
-        $arguments['pages'] = $this->quizTemplateService->getQuizNumber($requestAttributes);
-        $arguments['username'] = $this->quizTemplateService->getName();
-        $arguments['quizzes'] = $this->quizTemplateService->getQuizzes($requestAttributes, $request->getParameter('page'));
-//        $arguments['quizUser'] = $this->quizTemplateService->getQuizUser();
-
-
-        return $this->renderer->renderView("admin-quizzes-listing.phtml", $arguments);
+        $resultsPerPage = 5;
+        $quizName = $this->quizTemplateService->getFromParameter('quizName', $request, "");
+        $userId = $this->quizTemplateService->getFromParameter('userId', $request, "");
+        $currentPage = (int)$this->quizTemplateService->getFromParameter('page', $request, 1);
+        $totalResults = (int)$this->quizTemplateService->getEntityNumberOfPagesByField(QuizTemplate::class, ['name' => $quizName, 'user_id' => $userId]);
+        $quizzes = $this->quizTemplateService->getEntitiesByField(QuizTemplate::class, ['name' => $quizName, 'user_id' => $userId], $currentPage, $resultsPerPage);
+        $users = $this->quizTemplateService->getEntitiesByField(User::class, ['role' => 'Admin'], 1, 99999999);
+
+        $paginator = new Paginator($totalResults, $currentPage, $resultsPerPage);
+        $paginator->setTotalPages($totalResults, $resultsPerPage);
```

Comment:
why do you need that setter if you already passed the numbers in the construct?
and why does `setTotalPages` receive 2 arguments? It's a setter